src/raftstore/store/snap.rs

```diff
@@ -509,7 +509,7 @@ mod v1 {
                     let cap = cmp::min(self.left, DEFAULT_READ_BUFFER_SIZE);
                     let mut buf = vec![0; cap];
                     while self.left > 0 {
-                        try!(self.read(&mut buf));
+                        try!(self.read_exact(&mut buf));
```

Comment:
`read` here doesn't have to read all the content at once, the outside while loop ensures this already.